### PR TITLE
[FEAT] Added custom logger function

### PR DESCRIPTION
**Pull request by yubinkarki**
Added a module under the `utils` directory to create a slightly customized logger function.
Replaced the 4 print statements at the end of the main file with the logger.

How to use:

- Import the `setup_logger` function in the desired module
- Call the function with the name of the current module
- Use the logger with appropriate [loging level](https://docs.python.org/3/library/logging.html#logging-levels)

Screenshot:
image

### CODE REVIEW

File: utils/custom_logger.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""Custom logging module with some formatting."""
+
+import sys
+import logging
+
+
+def setup_logger(name: str) -> logging.Logger:
```
Comment (horrormyth) [inline utils/custom_logger.py:7]: Would it be better to use the logging dictconfig/yaml like so ? And then one can simply use `logger = getLogger(__name__)` when/wherever they need it. Do make changes to the dictconfig as required if you wanted to implement it.

```python

import logging
import logging.config

logging_config = {
    'version': 1,
    'formatters': {
        'standard': {
            'format': '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
        }
    },
    'handlers': {
        'file': {
            'class': 'logging.FileHandler',
            'filename': 'app.log',
            'level': 'DEBUG',
            'formatter': 'standard'
        },
        'console': {
            'class': 'logging.StreamHandler',
            'level': 'INFO',
            'formatter': 'standard',
            'stream': 'ext://sys.stdout'
        }
    },
    'loggers': {
        '': {
            'handlers': ['file', 'console'],
            'level': 'DEBUG',
            'propagate': True
        }
    }
}

logging.config.dictConfig(logging_config) #configure it during the startup

# Example usage
logger = logging.getLogger(__name__)
logger.debug('This is a debug message')
logger.info('This is an info message')
logger.warning('This is a warning message')
logger.error('This is an error message')
logger.critical('This is a critical message')

**Comment by NISH1001** [comment]
I'd recommend using [loguru](https://github.com/Delgan/loguru) for efficiency and avoiding any bugs on custom implementation.

**Comment by horrormyth** [comment]
> I'd recommend using [loguru](https://github.com/Delgan/loguru) for efficiency and avoiding any bugs on custom implementation.

I think Python builtin logging module is quite self sufficient for the current use case and it just needs the configuration. Also works across all platform, it's best to avoid additional package for the same. It might be a bit overkill

**Comment by yubinkarki** [comment]
Working on it.

**Comment by yubinkarki** [comment]
Changes made:
- Used `logging.config.dictConfig` to configure the logger with custom options.
- Called the logger config function at the top of `school_center.py` file.
- Error level logs are saved under project_root/logs/custom_logs.log _(ignored)_

Console logs:
image

Log file:
image

**Comment by yubinkarki** [comment]
I have added the mentioned changed. Please review @horrormyth.

**Review by horrormyth — APPROVED** [review]
LGTM

File: school_center.py [hunk 2]
```diff
@@ -204,8 +211,7 @@ def is_allocated(scode1: str, scode2:str) -> bool:
             print(f"{to_allot}/{s['count']} left for {s['scode']} {s['name-address']} centers: {len(centers_for_school)}")
```
Comment (sapradhan) [inline school_center.py:211]: use logger here as well at warn or error level

Comment (yubinkarki) [reply school_center.py]: @sapradhan Done.

**Pull request merged**